rest/data/host_create.go

```diff
@@ -157,6 +162,56 @@ func CreateHostsFromTask(ctx context.Context, env evergreen.Environment, t *task
 	return catcher.Resolve()
 }
 
+// GetHostCreateDistro returns the distro based on the name and provider.
+// If the provider is Docker, passing in the distro is required, and the
+// distro must be a Docker distro. If the provider is EC2, the distro
+// name is optional.
+func GetHostCreateDistro(ctx context.Context, createHost apimodels.CreateHost) (*distro.Distro, error) {
+	var d *distro.Distro
+	var err error
+	isDockerProvider := evergreen.IsDockerProvider(createHost.CloudProvider)
+	if isDockerProvider {
+		d, err = distro.FindOneId(ctx, createHost.Distro)
+		if err != nil {
+			return nil, errors.Wrapf(err, "finding distro '%s'", createHost.Distro)
+		}
+		if d == nil {
+			return nil, errors.Errorf("distro '%s' not found", createHost.Distro)
+		}
+		if !evergreen.IsDockerProvider(d.Provider) {
+			return nil, errors.Errorf("distro '%s' provider must support Docker but actual provider is '%s'", d.Id, d.Provider)
+		}
+	} else {
+		d = &distro.Distro{}
+		if createHost.Distro != "" {
+			var dat distro.AliasLookupTable
+			dat, err := distro.NewDistroAliasesLookupTable(ctx)
+			if err != nil {
+				return nil, errors.Wrap(err, "getting distro lookup table")
+			}
+			distroIDs := dat.Expand([]string{createHost.Distro})
+			if len(distroIDs) == 0 {
+				return nil, errors.Wrap(err, "distro lookup returned no matching distro IDs")
+			}
+			foundDistro, err := distro.FindOneId(ctx, distroIDs[0])
+			if err != nil {
+				return nil, errors.Wrapf(err, "finding distro '%s'", createHost.Distro)
+			}
+			if foundDistro == nil {
+				return nil, errors.Errorf("distro '%s' not found", createHost.Distro)
+			}
+			d = foundDistro
+		}
+	}
+	// Do not provision task-spawned hosts.
+	d.BootstrapSettings.Method = distro.BootstrapMethodNone
+	if !isDockerProvider {
+		// Set provider only if it's not a Docker provider.
+		d.Provider = evergreen.ProviderNameEc2OnDemand
```

Comment:
nit: rather than make a separate `if` block just for this, you can just absorb this into the `else` (i.e. EC2) case above.